### ISSUE TITLE
About Loss 

### ISSUE THREAD

**Issue by TINA-WA** [ZhiyuanDang/NNM#11]
SCANLoss.py中
“
similarity = torch.bmm(anchors_prob.view(2 * b, 1, n), positives_prob.view(2 * b, n, 1)).squeeze()
ones = torch.ones_like(similarity)
consistency_loss = self.bce(similarity, ones)

anchors_prob_c = F.normalize(anchors_prob, dim = 0)
positives_prob_c = F.normalize(positives_prob, dim = 0)

similarity = torch.mm(anchors_prob_c.t(), positives_prob_c)
labels = torch.tensor(list(range(n))).cuda()
ce_loss =  self.ce(similarity, labels)
”
如果consistency_loss对应文章中的L_local_consi
![image](https://user-images.githubusercontent.com/46915657/226889762-3bd666d5-c235-46e2-9ba1-3d72070db21d.png)

**Question1:**ce_loss是否对应文章中的L_local_class?
![image](https://user-images.githubusercontent.com/46915657/226890003-52363b3a-599b-4ee7-a578-5a9eb1b84005.png)

文章中定义L_local_class里的q是p的转置矩阵
![image](https://user-images.githubusercontent.com/46915657/226890579-bae591dc-8370-4ea9-92d4-8ed5a9035702.png)
这样qi即为原来p矩阵的列向量，列向量看成是一批次图片在某一类上的表现，class contrastive loss

**Question2:**但在代码中，并没有转置操作，是直接拿p代替了q来计算，不知道是我哪里理解错了，还是这里的q应该写为p?

**Comment by ZhiyuanDang** [comment]
Q1：是

Q2：已进行转置，标准contrastive loss是 torch.mm(anchors_prob, positives_prob.t())，如果是class contrastive loss的话，是如代码中这样torch.mm(anchors_prob_c.t(), positives_prob_c)。

**Comment by TINA-WA** [comment]
Thanks very much! Wish you all the best!
